### PR TITLE
UML-431 - Codecov added to pipeline

### PR DESCRIPTION
**Pull request by williamfalconeruk**
- All lcov uploads to same as php one
- Tweaks to use node for jest test instead of puppeteer
- Use node-browser docker image
- Added arch key for uniqueness in caching
- Additional steps for calling webhook on coveralls 
- Added docker images for php-cli-node for the upload
this is subject to the result of issue request:
https://github.com/lemurheavy/coveralls-public/issues/1390#issuecomment-594410482
- reverted to use codecov. 

## Purpose
_Briefly describe the purpose of the change, and/or link to the JIRA ticket for context_

Fixes UML-431

## Approach

- Changes to CircleCI process (running parallel with coveralls) 
 - **NOTE** both codecov and coveralls are running at this time.  
- Codecov's behaviour is set so that:
     -  it shows all of its data and edits the coverage comment.
     - It will always update that comment every time a change is made even if there is no change.

Next steps will be to: 
- switch off coveralls and and configure git to expect codecov to be completed.
- change the settings so it does 80% coverage as per coveralls.

## Learning

- Circle CI Build process
- Coveralls
- CodeCov.io

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -585,7 +585,7 @@ orbs:
 
       # Code Coverage
 
-      codecov_php-unit_upload:
+      codecov_upload:
```
Comment (cooperaj) [inline .circleci/config.yml]: Don't think codecov is even used at this point. This task definition just hasn't been taken out yet

**Comment by cooperaj** [comment]
Wait, codecov is working?

🕺

**Comment by williamfalconeruk** [comment]
> Wait, codecov is working?
> 
> 🕺

Yup! only. Thing is now it needs to add a PR comment when done. currently not possible as CodeCov is waiting for the build to complete, as it has an approval state. Once the build split is done in UML-540, this should theoretically work as i'd expect.

**Comment by williamfalconeruk** [comment]
Closing for now - will reinstate at a later date.

**Comment by williamfalconeruk** [comment]
Screenshot 2020-06-10 at 11 20 08 (2)
codecov coverage working now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -288,19 +310,21 @@ orbs:
         docker: [image: circleci/php:7-cli-browsers]
       node:
         docker: [image: circleci/node:14.0]
+      node-browsers:
+        docker: [image: circleci/node:14.0-browsers]
```
Comment (andrewpearce-digital) [inline .circleci/config.yml:314]: Do you think it's worth consolidating the node and node-browsers executors to all be node-browsers?

As it's a circleci image it'll be cached and quickly pulled anyway.

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+codecov:
+  bot: cooperaj
```
Comment (andrewpearce-digital) [inline codecov.yml:2]: How come we're calling this cooperaj? We could/should use opg-use-an-lpa (admin user on github)

Comment (williamfalconeruk) [reply codecov.yml:2]: lets try that out. i wasn't keen on using this anyway.

File: .circleci/config.yml (hunk 2 above)
Comment (williamfalconeruk) [reply .circleci/config.yml:314]: I'll take a look...

File: codecov.yml (hunk 3 above)
Comment (williamfalconeruk) [reply codecov.yml]: done.

File: .circleci/config.yml (hunk 2 above)
Comment (cooperaj) [reply .circleci/config.yml:314]: I the browsers one will start a number of background processes IIRC so if theres a cost/speed requirement it's probably better just to have different. Is it an issue cause the same could be said for the PHP side too

Comment (williamfalconeruk) [reply .circleci/config.yml:312]: Doesn't seem to be any issues with this as it stands. I've made the change, for now, I can always fix this back later if it does become a problem...

File: codecov.yml (hunk 3 above)
Comment (williamfalconeruk) [reply codecov.yml]: can't remember why I used adams one TBH...

**Pull request merged**